ambry-commons/src/main/java/com/github/ambry/compression/BaseCompression.java

```diff
@@ -222,6 +236,9 @@ public int compress(byte[] sourceBuffer, int sourceBufferOffset, int sourceDataS
     compressedBuffer[offset] = 1;
     offset++;
 
+    compressedBuffer[offset] = (byte) algorithmNameBinary.length;
+    offset++;
+
```

Comment:
should move these two line after line 243 to make sure that we called getAlgorithmNameBinary first.

Reply:
You are right.  Let me do that.